### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by micahl**
Linked the main README to the docs/README, fixed a few minor broken links, and added some additional intro with pointers to relevant conceptual docs.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -25,6 +27,9 @@ Each component talks to the other components via IPC in the form of HTTP-over-UD
 ![New component overview](img/new-component-overview-simple.svg)
 [(Click here for detailed version)](img/new-component-overview-detailed.svg)
 
+## Relation to IoT Edge
+
+The [IoT Edge runtime](https://docs.microsoft.com/azure/iot-edge/iot-edge-runtime) is responsible for interacting with a container engine to manage containerized IoT Edge modules. The code can be found in the [IoT Edge repo](https://github.com/azure/iotedge). It exposes APIs for those containerized Edge modules to request certificates, encrypt/decrypt secrets and obtain SAS tokens for talking to Azure IoT Hub. It implements these runtime operations by talking to the IS, CS and KS on behalf of the modules, ie it forwards these API requests to the identity, cert, and key services after enhancing them with the identity of the module making the request.
 
 # Provisioning and Runtime Operation
```
Comment (arsing) [inline docs/README.md:35]: The link on line 45 (GH won't let me add a review comment there) is also broken (try it on https://azure.github.io/iot-identity-service/ ), so please fix that too.

File: docs/identity-service.md [hunk 2]
```diff
@@ -3,6 +3,14 @@
 ## Get primary cloud identity for authenticated workload
 `GET /identities/identity`
```
Comment (arsing) [inline docs/identity-service.md:4]: All of these need the API version in the querystring.

`?api-version=2020-09-01`

File: docs/identity-service.md [hunk 3]
```diff
@@ -3,6 +3,14 @@
...
 
+The shape of the response will depend on the principal used to authenticate with the Identity Service. The association between an authorized principal and the identity type to return is based on the `idtype` in the identity service's [configuration](identity-service.md#module-provisioning--re-provisioning).
+
+The returned `keyHandle` is for use with the [keys service](keys-service.md). For `aziot`-type identities using SaS auth, `keyHandle` contains the key handle of the identity's primary shared access key, which can be used to generate a security token (for IoT Hub authentication) using the keys service [sign](keys-service.md#sign) operation.
```
Comment (arsing) [inline docs/identity-service.md:8]: `[keys service]` -> `[Keys Service]`

File: docs/identity-service.md [hunk 4]
```diff
@@ -3,6 +3,14 @@
 ## Get primary cloud identity for authenticated workload
 `GET /identities/identity`
 
+The shape of the response will depend on the principal used to authenticate with the Identity Service. The association between an authorized principal and the identity type to return is based on the `idtype` in the identity service's [configuration](identity-service.md#module-provisioning--re-provisioning).
+
+The returned `keyHandle` is for use with the [keys service](keys-service.md). For `aziot`-type identities using SaS auth, `keyHandle` contains the key handle of the identity's primary shared access key, which can be used to generate a security token (for IoT Hub authentication) using the keys service [sign](keys-service.md#sign) operation. 
+For `aziot`-type device identities using X.509 auth, `keyHandle` contains the key handle of the identity X.509 certificate's private key, used for IoT Hub client authentication during TLS handshake using an SSL engine.    
+
+`certId` contains the certificate ID of the identity X.509 certificate, which can be retrieved used using the [certificates service](certificates-service.md) while connecting to IoT Hub.
```
Comment (arsing) [inline docs/identity-service.md:11]: `[certificates service]` -> `[Certificates Service]`

File: docs/README.md [hunk 5]
```diff
@@ -1,12 +1,14 @@
 # Overview
 
-IoT Identity Service is made up of three services that provide provisioning and cryptographic services:
+Azure IoT Edge brings the analytical power of the cloud closer to devices to drive better business insights and enable offline decision making through the use of [Edge modules](https://docs.microsoft.com/azure/iot-edge/iot-edge-modules).  The full IoT Edge runtime, in concert with IoT Hub, [provides numerous capabilities](https://docs.microsoft.com/azure/iot-edge/iot-edge-runtime) to scale out and manage an IoT solution from the cloud.
```
Comment (arsing) [inline docs/README.md:3]: The rest of the docs don't use two spaces after the period, so please don't use them here.

File: README.md [hunk 6]
```diff
@@ -1,11 +1,16 @@
-This repository contains the code of the Azure IoT Identity Service and related services. Together these services make a basic device "runtime" for Azure IoT devices.
+# IoT Identity Services for Edge Devices
```
Comment (arsing) [inline README.md:1]: It's not just for Edge devices.

File: docs/README.md (hunk 5 above)
Comment (arsing) [inline docs/README.md:3]: Please don't start off by talking about IoT Edge. This is more fundamental than IoT Edge.

File: docs/README.md (hunk 1 above)
Comment (micahl) [reply docs/README.md:35]: I see the link is broken on the GitHub Pages version of the docs, but on [the repo ](https://github.com/Azure/iot-identity-service/tree/main/docs) it's working.  I'm not very familiar with Github Pages, but it seems like the _index.md files weren't included in the site while the other regular .md files were.

Comment (arsing) [reply docs/README.md:35]: Yes, because I think Jekyll wants them to be called `index.md`, not `_index.md`

Comment (micahl) [reply docs/README.md:35]: Ok, I'll rename.

File: README.md [hunk 7]
```diff
@@ -1,11 +1,16 @@
...
+# Architecture
+
+See [`docs/README.md`](docs/README.md) for an overview of these services and their relation to [`Azure IoT Edge`](https://github.com/Azure/iotedge).
 
 # Contributing
 
 See [`CONTRIBUTING.md`](CONTRIBUTING.md) for contribution guidelines.
 
-See [`docs/_index.md`](docs/_index.md) for developer documentation.
+See [`docs/dev/_index.md`](docs/dev/_index.md) for developer documentation.
```
Comment (massand) [inline README.md:13]:
```suggestion
See [`docs/dev/index.md`](docs/dev/index.md) for developer documentation.
```

File: docs/README.md [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
+Azure IoT Edge brings the analytical power of the cloud closer to devices to drive better business insights and enable offline decision making through the use of [Edge modules](https://docs.microsoft.com/azure/iot-edge/iot-edge-modules). The full IoT Edge runtime, in concert with IoT Hub, [provides numerous capabilities](https://docs.microsoft.com/azure/iot-edge/iot-edge-runtime) to scale out and manage an IoT solution from the cloud.
+
+The IoT Identity Service is one component of IoT Edge and can be used stand-alone. The "Identity Service" is technically made up of three services (purple in the diagram) that provide provisioning and cryptographic services. They provide the foundation for authorized components on the device to communicate with Azure services (i.e. IoT Hub) whether as the [device](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-identity-registry) or an individual [module](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-module-twins). Containerized modules (i.e. IoT Edge modules) require the full IoT Edge runtime. Non-containerized modules (i.e. host-level modules) minimally need the identity, cert, and key services described below.
```
Comment (massand) [inline docs/README.md:5]: In code, packaging and repo structure, IoT Edge depends on Identity Service. Identity Service is NOT a component of IoT Edge. A rephrase of the description for consideration - Identity Service for edge devices helps manage various types of IoT identities and enables crypto operations needed for various applications. Components within IoT Edge depend on the Identity Service.

File: docs/README.md [hunk 9]
```diff
@@ -1,12 +1,14 @@
...
 
 - [Identity Service](identity-service.md)
 
     This service provisions the device identity and module identities with Azure. 
 
-    The Identity Service provides access to host-level processes and container modules to connect to the cloud using tokens or X.509 certificate credentials, corresponding to their respective identities.
+    The Identity Service provides access to both host-level processes and container modules (via the IoT Edge runtime) to connect to the cloud using tokens or X.509 certificate credentials, corresponding to their respective identities.
```
Comment (massand) [inline docs/README.md:11]: I would just remove the distinction between processes and containers. Let us call them all "modules". IoT Edge currently doesn't manage host-level processes (only root user can), but that could change in the future.

**Review by massand — CHANGES_REQUESTED** [review]
Nit change. But, I agree with @arsing that we

File: docs/README.md (hunk 8 above)
Comment (micahl) [reply docs/README.md]: I've updated the wording and shifted a few things around to indicate the relationship between IoT Edge and the identity service.

File: docs/README.md (hunk 5 above)
Comment (micahl) [reply docs/README.md]: Revised wording.

File: README.md (hunk 6 above)
Comment (micahl) [reply README.md:1]: I'm taking an expansive view here of what is considered an "edge device".

**Pull request closed without merging**